internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadWriteSchemaRepository.java

```diff
@@ -486,7 +487,9 @@ public Pair<Integer, Integer> getDerivedSchemaId(String storeName, String derive
         .stream()
         .flatMap(List::stream)
         .collect(Collectors.toList())) {
-      if (derivedSchemaEntry.getSchema().equals(derivedSchema)) {
+      String repoSchemaStr = AvroCompatibilityHelper.toParsingForm(derivedSchemaEntry.getSchema());
+      String derivedSchemaStrToFind = AvroCompatibilityHelper.toParsingForm(derivedSchema);
```

Comment:
Can this move out of the loop, since it seems repetitive here?

Also, I wonder if we should overload the `String derivedSchemaStr` param into a `Schema` or perhaps even `SchemaEntry` param, so that the initial `String -> Schema` conversion need not happen? For example, I see that the call from the `MetaStoreWriter` is performing a `Schema -> String` conversion to pass into this API, which then does `String -> Schema -> canonicalized String`... That seems a bit unfortunate?

Reply:
hmm, the issue was in `get_value_or_derived_schema_id` call path which is just once during samza startup, did not realize its called in so many other places. Let me see if it can be optimized.  Also initial string and canonical form could still differ wrt docs/annotations. we could process it once and store the canonical form somewhere?

Reply:
It's not called from a ton of places, and does not seem like a hot path, but still, I get wary when I see lots of string parsings... For the controller path, it needs to use String as the starting point, but for `MetaStoreWriter`, my understanding is that we'll do `Schema -> String -> Schema -> canonicalized String` and I'm suggesting it could be shrunk down to just `Schema -> canonicalized String` without too much effort.

Reply:
actually the impl are using different classes, the MetaStoreWriter is using `HelixReadOnlySchemaRepository` vs the change in this PR is in `HelixReadWriteSchemaRepository` The write compute schema lookup from samza is sending a string, so changing the param to use Schema or SchemaEntry wont help much if we needed to convert the string->schema .